### PR TITLE
Initial commit for new membership policy page

### LINKED ISSUE #75: Need community membership policy document
Per [email](https://groups.google.com/g/kubevirt-dev/c/Md0tFvI-FYg?pli=1) to kubevirt-devel from <sgott@redhat.com>

```
Lately we've been thinking about the diversity of our project maintainers. It's easy to notice that a lot of emails come from one domain. When our project was in its infancy, that was understandable and sort of made sense. However, I'd like to think we've grown past that. We are starting to see more names on our community calls, and an ever growing number of followers on various platforms such as GitHub, Twitter or Slack. With that in mind, a topic I'd like to discuss during tomorrow's community meeting is the establishment of a documented process by which community members can get involved and become maintainers or co-owners of the KubeVirt project.

As a follow on to that, it's worth also discussing the different ways people interested in the project might be able to get involved. Some areas of responsibility might include:
 * Moderating a public GitHub issue scrub
 * Building the Community Meeting weekly agenda
 * Slack presence -- to answer questions of other community members
 * Leading or contributing towards our effort to get KubeVirt to V1
 
If you have any other ideas, please feel free to bring them up on this email thread or during the call!

Here's a draft proposal for what our maintainership procedures could look like:

https://docs.google.com/document/d/1sviYdFC5gXIRhc8M1nFOb1aDzyS-fmao9JjzSzdORTQ/edit#
https://docs.google.com/document/d/1N9SflndMOh9H7POvEMR__KWV--fi0e7OoTO76YcqAFs/edit
```

### PR DESCRIPTION
**Pull request by mazzystr**
Initial commit for new membership policy page

> Fixes #75

### CODE REVIEW

**Review by dhiller — APPROVED** [review]
/lgtm

File: membership_policy.md
Comment (rmohr) [inline membership_policy.md:38]: It is not clear to me how to get sponsored. What do I have to do? Create a PR in project-infra for peribolos and ask some maintainers to review?

Comment (mazzystr) [reply membership_policy.md:38]: This is a policy page not an implementation page.  

@dhiller created implementation documentation [here](https://github.com/kubernetes/test-infra/tree/master/prow/cmd/peribolos).

File: membership_policy.md
Comment (dhiller) [inline membership_policy.md:42]: I think @rmohr has a point above, maybe we can rephrase this into "Open a PR against the [org members section](https://github.com/kubevirt/project-infra/blob/master/github/ci/prow/files/orgs.yaml#L21)" ?

File: membership_policy.md
Comment (mazzystr) [reply membership_policy.md:38]: @dhiller @rmohr  Are we good to resolve this point with the update to L42?

Comment (rmohr) [reply membership_policy.md:38]: Much better now with the changes. :+1:

**Review by dhiller — APPROVED** [review]
/lgtm
/approve
/hold

I think we are good for a first version. But let's wait a day for feedback from others.

File: membership_policy.md
Comment (jberkus) [inline membership_policy.md:67]: s/code/code, documentation, and other contributions/

We really need documentation contributors, and people helping manage the project.  These folks are legit contributors as much as folks who write code.  We need to have, and promote, documentation-only reviewers.

Comment (mazzystr) [reply membership_policy.md:67]: Thanks for moving the conversation here for the sake of public transparency and discourse.

Would you accept s/code/contribution/g to be less wordy?

Comment (mazzystr) [reply membership_policy.md:67]: @EdDev writes re `some part of the project. `
```
This is repeated multiple times but has no definition. I mean, the "parts" should be specified in advance, perhaps by grouping. It may not currently fit in dedicated OWNERS files, but at least one can examine to which group the reviewer is assigned and what are these parts domain.
```

File: membership_policy.md
Comment (mazzystr) [inline membership_policy.md:80]: @EdDev writes ...
```
For both reviewers and approvals, I would suggest also a mentoring period to assure one is ready for the task. This may be a requirement from the sponsor, which may be natural, but it may be done by any approver I guess.
```

Comment (mazzystr) [reply membership_policy.md:80]: @EdDev Can you suggest a method of monitoring that we can document?  Imo, by the time we are to this point a review / approvers reputation is well established.  I'm not sure about other but I definitely don't have time to be babysitting.

File: membership_policy.md
Comment (mazzystr) [inline membership_policy.md:83]: @EdDev writes ...
```
IMO it should be nominated by a sponsor. Self-nomination is odd, especially if done in public. It is the sponsor who needs to check in advance that it is agreed by others who may object, avoiding awkward situations.
```

@stu-gott writes ...
```
I think the list of requirements above is pretty clear. I don't think we'll run into situations where we need to say no without good reason.
```

@jberkus writes ...
```
agreed that people rarely self-nominate for this.  THis does bring up needing a regular cadence on having our reviewers/approvers regularly review potential candidates.
```

File: membership_policy.md
Comment (mazzystr) [inline membership_policy.md:99]: @dankenigsberg writes ...
```
I see that we have "emeritus_approvers". How an approver becomes one? What does it mean?
```

@stu-gott writes ...
```
That's a good point. Those in that category used to be active members of the project, but have moved on for various reasons. @rmohr@redhat.com might have insight as to what effect being in emeritus status has for the purposes of the owners file ("git blame" indicates he's edited that section).
```

@rmohr writes ...
```
prow will suggest contributors once a PR went through inital review and looks too, to add an approver to the PR. It will suggest people based on the owners file. I therefore clean up the owners file from time to time, so that the suggestions make sense.

Approvers which left the project or don't contribute often enough anymore are moved to the emeritus section for two reasons:
* current approvers can see who may have insight about things which were done in the past
* to honor their past contributions
```

File: membership_policy.md
Comment (mazzystr) [inline membership_policy.md:103]: @oshoval writes ...
```
Should it be good to keep a txt file beside the OWNERS, which tells which approve responsible to which part(s)?
i.e storage, networking, virtualization, etc

or we don't want to maintain such file
```

@stu-gott writes ...
```
There's OWNER files in sub directories -- I think the scope is implicit.
```

File: membership_policy.md
Comment (mazzystr) [inline membership_policy.md:111]: @EdDev writes ...
```
I have not seen a definition of this role here. Is this something new?
I thought  the approvers are the maintainers.
```

@stu-gott writes ...
```
typo
```

File: membership_policy.md
Comment (mazzystr) [inline membership_policy.md:125]: @jberkus writes ...
```
Maintainer/Owner

don't we need a definition for maintainer/subproject owner here?  how does someone get to be a maintainer?  or are there none?

difference between approver and maintainer usually: maintainers have owner in the root of the repos
```

File: membership_policy.md
Comment (mazzystr) [reply membership_policy.md:67]: Thanks for moving the conversation over here for the sake of public transperancy.

Would you accept s/code/contribution/g to be less wordy?

File: membership_policy.md
Comment (jberkus) [reply membership_policy.md:41]: Also: FWIW, Kubernetes doesn't actually enforce the "must be a reviewer" requirement these days.  The automation actually only checks that the two sponsors are org members.

File: membership_policy.md
Comment (jberkus) [reply membership_policy.md:37]: Definition used by Kubernetes is "at least 50 github actions in the last year".  Does that work?

File: membership_policy.md
Comment (mazzystr) [reply membership_policy.md:111]: @EdDev and @stu-gott  I'm resolving this one.

**Comment by dhiller** [comment]
@mazzystr we would need to update the [CONTRIBUTING.md](https://github.com/kubevirt/kubevirt/blob/master/CONTRIBUTING.md#becoming-a-member) section with a link to the documents here

**Comment by codificat** [comment]
> @mazzystr we would need to update the [CONTRIBUTING.md](https://github.com/kubevirt/kubevirt/blob/master/CONTRIBUTING.md#becoming-a-member) section with a link to the documents here

yes, and also the [contributing section in the user guide](https://github.com/kubevirt/user-guide/blob/master/docs/appendix/contributing.md)

**Comment by dhiller** [comment]
@mazzystr is there still something to do here? Or can we merge this, so that we can point people towards the documents here?

**Comment by mazzystr** [comment]
@rmohr  @fabiand  I think we're about ready for one last review here.  I would like to have this in place end-of-this-week or first-day-next-week.  

Per @jberkus There is still a lot to talk about but let's do it via an issue for the sake of something is better than nothing.

**Review by rmohr — APPROVED** [review]

File: membership_policy.md
Comment (davidvossel) [inline membership_policy.md:34]: to/too ?

**Review by davidvossel — COMMENTED** [review]
seems reasonable. found one minor thing maybe

File: membership_policy.md
Comment (EdDev) [inline membership_policy.md:10]: There seems to be an additional column here. `Sponsored by 2 approvers` appears in the wrong one.

File: membership_policy.md
Comment (EdDev) [inline membership_policy.md:71]: Like in the approver case, a `Reviewer Status is scoped to a part of the project.` can be added here as well.

File: membership_policy.md
Comment (EdDev) [inline membership_policy.md:119]: What is `L+`?

Comment (mazzystr) [reply membership_policy.md:119]: An IDE command, LOL!  I'll fix it.  :)

File: membership_policy.md
Comment (EdDev) [inline membership_policy.md:10]: nit: It does not appear as a bullet now.

**Review by EdDev — APPROVED** [review]
Thank you, looks great.
Hope we can get it in soon, we can fix other issues later on.

**Comment by EdDev** [comment]
@mazzystr , can we get this in? There is one nit, but it can get in as is IMO.

**Pull request merged**
